File: 0901-1000/0910-Smallest Range II/0910-Smallest Range II.py
class Solution:
    def smallestRangeII(self, A: List[int], K: int) -> int:
        A.sort()
        result = A[-1] - A[0]
        for i in range(len(A) - 1):
            result = min(result, max(A[i] + K, A[-1] - K) - min(A[i + 1] - K, A[0] + K))
        return result

File: 0901-1000/0910-Smallest Range II/0910-Smallest Range II.cpp
class Solution {
public:
    int smallestRangeII(vector<int>& A, int K) {
        sort(A.begin(), A.end());
        int minNum = A[0], maxNum = A.back();
        int result = maxNum - minNum;
        for (int i = 0; i < A.size() - 1; ++i) {
            result = min(result, max(A.back() - K, A[i] + K) - min(A[0] + K, A[i + 1] - K));
        }
        return result;
    }
};
